### ISSUE TITLE
Same-site cookie support (feature request)

### ISSUE THREAD

**Issue by lukebailey** [craftcms/cms#4462]
### Description
Feature request to add support for same site cookies which is now available in PHP 7.3: https://wiki.php.net/rfc/same-site-cookie. 

To prevent CSRF attacks this should be the default for all Craft cookies and available for plugin developers.

There is also an open ticket with Yii2 (https://github.com/yiisoft/yii-web/issues/61), I assume this is a dependancy on Craft adopting the feature?

**Comment by lukebailey** [comment]
After a discussion with Alexander Makarovex it appears this is now available since version 2.0.21

https://github.com/yiisoft/yii2/commit/1ed6ec1e5cdcd0bacbda67a129efa1aa88648d66

**Comment by brandonkelly** [comment]
Amazing! Just added a new `sameSiteCookieValue` for the next release.

To get the change early, change your `craftcms/cms` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#0bdc637564515165fe85d58e3e8f966c3d995572 as 3.1.32",
  "...": "..."
}
```

Then run `composer update`.

**Comment by OwenMelbz** [comment]
@brandonkelly Are you able to point us in the direction to the docs to show how to use this feature?

**Comment by brandonkelly** [comment]
Yep, that looks right.

**Comment by karlmacklin** [comment]
How can we solve this for Craft 2?
I know it's not receiving updates anymore, other than security updates. Is this just in between 'regular' and security?

Our option is to 'hack core', edit yiilite.php maybe and add the $options with SameSite to php's internal session_set_cookie_params function call.
With risk of it being overwritten from future CMS update, depending on what future updates will bring.

Any suggestion on what to do? We currently have to set SameSite to None.

**Comment by brandonkelly** [comment]
@karlmacklin I just released Craft 2.9.2 with a new [sameSiteCookieValue]() config setting, to match the one in Craft 3.
